include/zephyr/logging/log_core.h

```diff
@@ -197,7 +197,7 @@ static inline char z_log_minimal_level_to_char(int level)
  * emitted instead. String check may increase compilation time so it is not
  * always performed (could significantly increase CI time).
  */
-#if CONFIG_LOG_FMT_STRING_VALIDATE
+#if defined(CONFIG_LOG_FMT_STRING_VALIDATE) && CONFIG_LOG_FMT_STRING_VALIDATE
```

Comment:
it should be enough to have:
```
#ifdef CONFIG_LOG_FMT_STRING_VALIDATE
```

It is used that way across the repo and any `#if` were changed to `#ifdef`s for kconfig bools.